Tools/Scripts/run-mvt-tests

```diff
@@ -30,317 +30,350 @@ if os.path.isdir(os.path.join(scripts_dir, 'webkitpy')):
+            _log.info(f"    - {expected_failures} tests were expected failures.")
+        if unexpected_failures > 0:
+            _log.info(f"    - {unexpected_failures} tests were unexpected failures.")
+        _log.info('-' * len(suite_summary_header_str) + '\n')
+
+    # Print global summary if more than one suite was run
+    if num_suites > 1:
+        _log.info('######################')
+        _log.info('# TOTAL TEST SUMMARY #')
+        _log.info('######################')
+        total_tests_run = 0
+        total_unexpected_failures = 0
+        total_expected_failures = 0
+        for suite in test_results:
+            total_tests_run += test_results[suite][0]
+            total_unexpected_failures += test_results[suite][1]
+            total_expected_failures += test_results[suite][2]
+        total_tests_pass = total_tests_run - total_unexpected_failures - total_expected_failures
+        suites_str = ", ".join(suites)
+        _log.info(f"Executed {num_suites} MVT test suites: {suites_str}")
+        _log.info(f"Ran {total_tests_run} tests in total of which:")
+        _log.info(f"    - {total_tests_pass} tests passed.")
+        if total_expected_failures > 0:
+            _log.info(f"    - {total_expected_failures} tests were expected failures.")
+        if total_unexpected_failures > 0:
+            _log.info(f"    - {total_unexpected_failures} tests were unexpected failures.")
+        exit_code = total_unexpected_failures
+
+    mvtresultsexpectations_parser.maybe_update_expectations()
+    return exit_code
+
+
+def configure_logging(selected_log_level='info'):
+    class LogHandler(logging.StreamHandler):
+        def __init__(self, stream):
+             super().__init__(stream)
+
+        def format(self, record):
+            if record.levelno > LOG_MESSAGE:
+                return '%s: %s' % (record.levelname, record.getMessage())
+            return record.getMessage()
+
+    logging.addLevelName(LOG_MESSAGE, 'MESSAGE')
+    if selected_log_level == 'debug':
+        log_level = logging.DEBUG
+    elif selected_log_level == 'info':
+        log_level = logging.INFO
+    elif selected_log_level == 'quiet':
+        log_level = logging.NOTSET
+    elif selected_log_level == 'minimal':
+        log_level = logging.getLevelName(LOG_MESSAGE)
+
+    handler = LogHandler(sys.stdout)
+    logger = logging.getLogger()
+    logger.addHandler(handler)
+    logger.setLevel(log_level)
+    return handler
+
+
+# This monkey-patches urllib3.request to ensure that it runs always with
+# a timeout. Otherwise sometimes selenium hangs indefinitively.
+def set_urllib3_request_default_timeout(timeout):
+    def patched_request(self, method, url, *args, **kwargs):
+        # Only set timeout if not already specified
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = timeout
+        return original_request(self, method, url, *args, **kwargs)
+    original_request = urllib3.request.RequestMethods.request
```

Comment:
I'm somewhat surprised that it's not needed to put this line before the `patched_request` is defined. Could you move it up to better indicate the intention even if it currently works?

Reply:
AFAIK Python functions can access variables declared above its own scope, it doesn't matter in the line when you declare the variable, what matters is if the variable is already defined when you call the function.
But I can declare it above the function declaration if that makes the code more clear.